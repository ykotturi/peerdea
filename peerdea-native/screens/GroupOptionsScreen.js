import * as WebBrowser from 'expo-web-browser';
import React, {Component} from 'react';
import {
  Button,
  Image,
  Platform,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  TouchableHighlight,
  View,
  Dimensions,
} from 'react-native';
import AsyncStorage from "@react-native-async-storage/async-storage";
import styles from '../components/style';

const { width, height } = Dimensions.get('window');

export default class HomeScreen extends React.Component {
  state = {
    username: ''
  };

  async setData() {
    let values
    try {
      values = await AsyncStorage.multiGet(['username']);
      this.setState({
        username: values[0][1],
      })
    } catch (err) {
      console.log(err);
    }
  };

  componentDidMount() {
    this.focusListener = this.props.navigation.addListener(
      'focus',
      () => { this.setData() },
    );
  };

  componentWillUnmount() {
    this.focusListener();
  };

  render() {
    const {navigate} = this.props.navigation;

    return (
      <View style={styles.container}>
          <Image
            style={{width: 300, height: 51}}
            source={require('../assets/images/peerdea-logo-draft.png')}
            accessible={true}
            accessibilityLabel="Peerdea app logo"
          />

          <Text style={styles.getStartedText}>Welcome to Peerdea, {this.state.username}!</Text>

          <View style={{width: '80%', justifyContent: 'center', alignItems: 'center'}}>
          <TouchableHighlight
           style={styles.buttonLargeLeft}
           onPress={() => navigate('CreateGroup')}
          >
            <Text style={styles.buttonText}>
              Create a new group
            </Text>
          </TouchableHighlight>
          <View style={{marginBottom: 5}} />
          <TouchableHighlight
           style={styles.buttonLargeLeft}
           onPress={() => navigate('JoinGroup')}
          >
            <Text style={styles.buttonText}>
              Join an existing group
            </Text>
          </TouchableHighlight>
          </View>
          </View>
    );
  }
}
