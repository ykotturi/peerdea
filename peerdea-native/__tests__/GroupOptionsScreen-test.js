import React from 'react';
import renderer from 'react-test-renderer';

import GroupOptionsScreen from '../screens/GroupOptionsScreen';

jest.mock('@react-native-async-storage/async-storage', () =>
  require('@react-native-async-storage/async-storage/jest/async-storage-mock')
);

describe('Group options Screen', () => {
  jest.useFakeTimers();

  it(`renders the group options screen`, () => {
    const mockNavigation = {
      navigate: jest.fn(),
      addListener: jest.fn().mockImplementation((event, callback) => {
        callback();
        //returning value for `navigationSubscription`
        return { remove: jest.fn() }
      }),
    };

    const tree = renderer.create(
      <GroupOptionsScreen navigation={mockNavigation} />);
    expect(tree.toJSON()).toMatchSnapshot();
    
    var textList = tree.root.findAllByType('Text')
    
    textList.forEach((el) => {
      if (el.props.children === 'Create a new group') {
        el.parent.parent.props.onClick()
        expect(mockNavigation.navigate).toHaveBeenCalledWith('CreateGroup')
      }
      else if (el.props.children === 'Join an existing group') {
        el.parent.parent.props.onClick()
        expect(mockNavigation.navigate).toHaveBeenCalledWith('JoinGroup')
      }
    })
  });
});
