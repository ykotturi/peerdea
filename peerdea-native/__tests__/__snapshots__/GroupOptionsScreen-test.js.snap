// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group options Screen renders the group options screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "paddingBottom": 30,
      "paddingTop": 18,
    }
  }
>
  <Image
    accessibilityLabel="Peerdea app logo"
    accessible={true}
    source={1}
    style={
      Object {
        "height": 51,
        "width": 300,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "rgba(96,100,109, 1)",
        "fontSize": 17,
        "lineHeight": 20,
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to Peerdea, 
    !
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "width": "80%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#0076ff",
          "borderColor": "#0076ff",
          "borderRadius": 9,
          "borderWidth": 1,
          "elevation": 3,
          "height": 50,
          "justifyContent": "center",
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "justifyContent": "center",
            "paddingHorizontal": "8%",
          }
        }
      >
        Create a new group
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 5,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#0076ff",
          "borderColor": "#0076ff",
          "borderRadius": 9,
          "borderWidth": 1,
          "elevation": 3,
          "height": 50,
          "justifyContent": "center",
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "color": "#ffffff",
            "fontSize": 20,
            "justifyContent": "center",
            "paddingHorizontal": "8%",
          }
        }
      >
        Join an existing group
      </Text>
    </View>
  </View>
</View>
`;
